tests/virtualization/universal/upgrade_guests.pm

```diff
@@ -22,6 +22,10 @@ sub run {
 
     script_run("mkdir /root/update_guests");
     foreach my $guest (keys %virt_autotest::common::guests) {
+        # Lock guest to prevent multiple tests run in parallel by checking lock_guest file
+        assert_script_run("ssh root\@$guest touch lock_guest");
+        my $check_guest_lock = "grep -q locked lock_guest && exit 1 || echo locked > lock_guest; exit \$?";
+        script_retry("ssh root\@$guest \"flock lock_guest sh -c '$check_guest_lock'\"", delay => 60, retry => 40, fail_message => "Guest is currently busy");
```

Comment:
For multiple guests case, with the foreach guests loop above,  this may create dead lock. Is it correct that our test only has 1 guest per job, so it can be perfectly avoided?

Reply:
I think, keeping the foreach is to keep the *capability* of the test file to directly run multiple guests if needed, just like other xen/kvm test modules. You know, it is hard to promise that test requirements always keep the same.